File: python/examples/quickstart_agent.py
"""
AgentGatePay Python SDK - Agent Quickstart Example

Time to first payment: ~5 minutes
"""

import os
from agentgatepay_sdk import AgentGatePay

def main():
    # Initialize client
    client = AgentGatePay(
        api_key=os.getenv("AGENTPAY_API_KEY"),  # Optional
        agent_id="my-ai-agent",
        debug=True
    )

    print("=== AgentGatePay Agent Quickstart ===\n")

    # Step 1: Issue a mandate
    print("Step 1: Issuing mandate...")
    mandate = client.mandates.issue(
        subject="my-ai-agent@example.com",
        budget=100,  # $100 USD
        scope="*",
        ttl_minutes=1440  # 24 hours
    )

    print("Mandate issued!")
    print(f"  Token: {mandate['mandateToken'][:50]}...")
    print(f"  Expires: {mandate['expiresAt']}\n")

    # Step 2: Submit payment with existing transaction
    print("Step 2: Submitting payment...")
    payment = client.payments.submit_tx_hash(
        mandate=mandate["mandateToken"],
        tx_hash="0x1234567890abcdef..."  # Your blockchain transaction hash
    )

    print("Payment submitted!")
    print(f"  Status: {payment.get('status')}")
    print(f"  Amount: ${payment.get('amountUsd')}")
    print(f"  Budget remaining: ${payment.get('budgetRemaining')}\n")

    print("✓ Complete! Your AI agent can now make payments.\n")


if __name__ == "__main__":
    main()

File: javascript/examples/quickstart-agent.ts
/**
 * AgentGatePay SDK - Agent Quickstart Example
 *
 * This example shows how an AI agent can:
 * 1. Issue a mandate for delegated spending
 * 2. Submit a payment using an existing transaction
 *
 * Time to first payment: ~5 minutes
 */

import { AgentGatePay } from '../src';

async function main() {
  // Initialize client (API key is optional but recommended for production)
  const client = new AgentGatePay({
    apiKey: process.env.AGENTPAY_API_KEY, // Optional: 'pk_live_...'
    agentId: 'my-ai-agent',
    debug: true // Enable debug logging
  });

  console.log('=== AgentPay Agent Quickstart ===\n');

  // Step 1: Issue a mandate
  console.log('Step 1: Issuing mandate...');
  const mandate = await client.mandates.issue(
    'my-ai-agent@example.com', // Subject (your agent ID)
    100,                         // Budget: $100 USD
    '*',                         // Scope: all resources
    1440                         // TTL: 24 hours
  );

  console.log('Mandate issued!');
  console.log(`  Token: ${mandate.mandateToken.substring(0, 50)}...`);
  console.log(`  Expires: ${new Date(mandate.expiresAt * 1000).toISOString()}\n`);

  // Step 2: Submit payment with existing blockchain transaction
  console.log('Step 2: Submitting payment...');
  const payment = await client.payments.submitTxHash(
    mandate.mandateToken,
    '0x1234567890abcdef...' // Your blockchain transaction hash
  );

  console.log('Payment submitted!');
  console.log(`  Status: ${payment.status}`);
  console.log(`  Amount: $${payment.amountUsd}`);
  console.log(`  Budget remaining: $${payment.budgetRemaining}\n`);

  console.log('✓ Complete! Your AI agent can now make payments.\n');
}

main().catch(console.error);
